### PR TITLE
Properly link system frameworks and weak frameworks into dynamic framework targets.

### LINKED ISSUE #8493: fails to links due to unknown symbols on >= 1.6.0.rc.1, success on .beta.2
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

cocoapods >= 1.6.0.rc.1 fails to link Texture

## What did you do?

When I've built the project, I have received the following output:

```
Ld /Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/Texture/AsyncDisplayKit.framework/AsyncDisplayKit normal x86_64 (in target: Texture)
    cd /Users/vitaly/projects/radiofx-ios-v4.0/Pods
    export IPHONEOS_DEPLOYMENT_TARGET=9.0
    /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch x86_64 -dynamiclib -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.2.sdk -L/Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/Texture -F/Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/Texture -F/Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/PINCache -F/Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/PINOperation -F/Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/PINRemoteImage -filelist /Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/Texture.build/Objects-normal/x86_64/AsyncDisplayKit.LinkFileList -install_name @rpath/AsyncDisplayKit.framework/AsyncDisplayKit -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -mios-simulator-version-min=9.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/Texture.build/Objects-normal/x86_64/AsyncDisplayKit_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -stdlib=libc++ -fobjc-arc -fobjc-link-runtime -lc++ -framework Foundation -framework PINRemoteImage -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/Texture.build/Objects-normal/x86_64/AsyncDisplayKit_dependency_info.dat -o /Users/vitaly/projects/radiofx-ios-v4.0/DerivedData/RadioFx/Build/Products/Debug-iphonesimulator/Texture/AsyncDisplayKit.framework/AsyncDisplayKit

Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_ALAssetsLibrary", referenced from:
      objc-class-ref in ASMultiplexImageNode.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Same thing works on 1.6.0.beta.2
Tried to build on xcode 10.1 and 10.2 beta - same result

### PR DESCRIPTION
**Pull request by dnkoutso**
closes #8493

Verified consuming the pod in the issue properly works.

Integration specs: https://github.com/CocoaPods/cocoapods-integration-specs/pull/213

### CODE REVIEW

**Comment by paulb777** [comment]
Is #8204 still fixed?

**Comment by dnkoutso** [comment]
Will verify.

**Comment by dnkoutso** [comment]
@paulb777 it should be but i dont see a sample to verify it. This change is not about vendored libraries or frameworks, more about system ones.

File: spec/unit/target/build_settings/pod_target_settings_spec.rb [hunk 1]
```diff
@@ -78,16 +78,16 @@ class BuildSettings
             @xcconfig.to_hash['OTHER_LDFLAGS'].should.include('-no_compact_unwind')
           end
 
-          it 'does include the libraries for the specifications' do
+          it 'does include the system libraries for the specifications' do
```
Comment (dnkoutso) [inline spec/unit/target/build_settings/pod_target_settings_spec.rb]: tests here use a dynamic framework target above:
```ruby
            @pod_target = fixture_pod_target(@spec, true) # the true means requires frameworks...
```

**Pull request merged**